pkg/api/types.go

```diff
@@ -1834,15 +1834,8 @@ func (p *Properties) IsNvidiaDevicePluginCapable() bool {
 // SetCloudProviderRateLimitDefaults sets default cloudprovider rate limiter config
 func (p *Properties) SetCloudProviderRateLimitDefaults() {
 	if p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket == 0 {
-		if p.HasVMSSAgentPool() && !p.IsHostedMasterProfile() {
-			var rateLimitBucket int
-			for _, profile := range p.AgentPoolProfiles {
-				if profile.AvailabilityProfile == VirtualMachineScaleSets {
-					rateLimitBucket += common.MaxAgentCount
-				}
-
-			}
-			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = rateLimitBucket
+		if p.AnyAgentUsesAvailabilitySets() || p.AnyAgentUsesVirtualMachineScaleSets() {
```

Comment:
Thanks for simplifying the per-agent pool multiplier :)

Let's just get rid of the `if p.AnyAgentUsesAvailabilitySets() || p.AnyAgentUsesVirtualMachineScaleSets() {` altogether as there's no other kind of agent pool. We can also retire the `DefaultKubernetesCloudProviderRateLimitBucket` const if it's no longer being used.

Reply:
@jackfrancis : Sure we can do that. Only reason I didn't remove as was not sure if empty AgentPoolProfiles supported (load to default 3 Linux node deployment) ?.

Reply:
Perhaps we can accommodate that by keeping the default const in the event that we have zero pools (and thus try to set the rate to zero)

Reply:
I didn't get what you meant above.
Is below snippet what you are looking for where we set default value when agent poll is count is 0 ?.
		var agentPoolProfileCount int
		agentPoolProfileCount = len(p.AgentPoolProfiles)
		if agentPoolProfileCount == 0 {
			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = DefaultKubernetesCloudProviderRateLimitBucket
		} else {
			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = agentPoolProfileCount * common.MaxAgentCount

		}

Reply:
yes

Reply:
Updated the PR with the changes.